cirq-ft/cirq_ft/algos/arithmetic_gates.py

```diff
@@ -222,7 +222,15 @@ class SingleQubitCompare(infra.GateWithRegisters):
 
     @cached_property
     def signature(self) -> infra.Signature:
-        return infra.Signature.build(a=1, b=1, less_than=1, greater_than=1)
```

Comment:
Should BiQubitsMixer be updated too?

Reply:
I think so, but I'd recommend to open a separate issue in Qualtran. I haven't looked at the decomposition too closely, but I'm not sure if `ancilla` doesn't register doesn't need to be part of the signature at all or whether it should be a directional register like the target of an `And` gate. 

Maybe @NoureldinYosri can clarify this on the separate issue and then we can send a PR to update it and it's usage. Also, it looks like we are not really using `BiQubitsMixer(adjoint=True)` anywhere.

Reply:
Ok. I'll bump by cirq version locally and see. This is good to merge.

Reply:
the Adjoint=True is used during the decomposition I believe.

![Screenshot 2023-10-09 at 4 13 04 PM](https://github.com/quantumlib/Cirq/assets/12097876/8abccba4-03e1-41c9-91c1-9df3710f0c3c)

Reply:
@tanujkhattar @fdmalone it should be directional like `And`